confdb/aclocal_cc.m4

```diff
@@ -696,6 +696,8 @@ dnl	eight
 dnl
 dnl In addition, a "Could not determine alignment" and a "error!"
 dnl return is possible.  
+dnl 
+dnl Note: this macro assumes AC_CHECK_SIZEOF([long long]) has been run
```

Comment:
We can run `AC_CHECK_SIZEOF([long long])` again inside this macro, instead of assuming that it has already been run.

Reply:
Btw, this is not a refactoring patch.  Refactoring generally refers to moving existing code with no changes to algorithmic logic.  Changing `#ifdef HAVE_LONG_LONG_INT` to `#if SIZEOF_LONG_LONG > 0` is a change in algorithmic logic, and hence not refactoring.  The comment message needs to be changed.

Reply:
In the current code, probably for legacy reasons, we allow disabling the support of `long long` or `long double`. And for the same reasons that user may disable the `long long/long double`, they may wish to specify the size -- for example, cross-compilation where host system disagree with build system. To support that, we need a central point to do the size checking. By setting cache values, having multiple `AC_CHECK_SIZEOF` may still work. However, I thought to have a logical stage in configure where we check compiler, then check basic data type sizes, and then alignment, will make the structure of `configure` easier to work with.

I agree it is beyond a typical refactoring for this PR. I'll revise the descriptions.

Reply:
Note that we started requiring C99 quite recently.  So, all the additional checks for C99 types should be removed (or an abort added if those are not available).

Reply:
Ok.